tools/slicec-cs/src/generators/enum_generator.rs

```diff
@@ -72,6 +141,205 @@ fn enum_values(enum_def: &Enum) -> CodeBlock {
     code
 }
 
+fn enumerators_as_nested_records(enum_def: &Enum) -> CodeBlock {
+    let mut code = CodeBlock::default();
+    let namespace = enum_def.namespace();
+
+    for enumerator in enum_def.enumerators() {
+        let escaped_identifier = enumerator.escape_identifier();
+        let mut builder = ContainerBuilder::new(
+            "public partial record class", // Dunet nested records are always public
+            &escaped_identifier,
+        );
+
+        if let Some(summary) = enumerator.formatted_doc_comment_summary() {
+            builder.add_comment("summary", summary);
+        }
+        builder
+            .add_comments(enumerator.formatted_doc_comment_seealso())
+            .add_obsolete_attribute(enumerator)
+            .add_base(enum_def.escape_identifier())
+            .add_fields(enumerator.associated_fields().unwrap_or_default().as_slice());
```

Comment:
I suspect this works:
```suggestion
            .add_fields(&enumerator.associated_fields().unwrap_or_default());
```
explicitly calling `as_slice` is often unnecessary. The compiler is generally good at coercing things to slices.